File: problems/2873-maximum-value-of-an-ordered-triplet-i/solution.py
# Link: https://leetcode.cn/problems/maximum-value-of-an-ordered-triplet-i/
# Level: Easy
# Date: 2025-04-02

class Solution:
    def maximumTripletValue(self, nums: List[int]) -> int:
        n = len(nums)
        res = 0
        for k in range(2, n):
            m = nums[0]
            for j in range(1, k):
                num1 = m
                num2 = nums[j]
                num3 = nums[k]
                res = max(res, ((num1 - num2) * num3))
                m = max(m, nums[j])
        return res

File: problems/2873-maximum-value-of-an-ordered-triplet-i/solution.cpp
// Link: https://leetcode.cn/problems/maximum-value-of-an-ordered-triplet-i/
// Level: Easy
// Date: 2025-04-02

class Solution {
    public:
        long long maximumTripletValue(vector<int>& nums) {
            long long res = 0;
            int n = nums.size();
            for (int i = 0; i < n; i++) {
                for (int j = i + 1; j < n; j++) {
                    for (int k = j + 1; k < n; k++) {
                        long long num1 = nums[i];
                        long long num2 = nums[j];
                        long long num3 = nums[k];
                        res = max(res, ((num1 - num2) * num3));
                    }
                }
            }
            return res;
        }
};